conman/routes/tests/test_models.py

```diff
@@ -44,6 +44,71 @@ def test_unique_url(self):
             Route.objects.create(url='/')
 
 
+class RouteGetAncestorsTest(TestCase):
+    """
+    Test Route().get_ancestors().
+
+    All of these tests assert use of only one query:
+        * Get the ancestors of a Route:
+
+            SELECT "routes_route"."id",
+                   "routes_route"."polymorphic_ctype_id",
+                   "routes_route"."url"
+              FROM "routes_route"
+             WHERE (NOT ("routes_route"."id" = 42)
+                    AND "routes_route"."url" IN ('/', '/route43/'))
+          ORDER BY "routes_route"."url" ASC
+    """
+    def test_just_created(self):
+        """Until saved, return empty Queryset."""
+        route = RouteFactory.build()
+
+        with self.assertNumQueries(0):
+            # Presumed nonsense as unsaved, so no query.
+            ancestors = list(route.get_ancestors())
```

Comment:
One possibility is [`FieldError`](https://docs.djangoproject.com/en/1.10/ref/exceptions/#django.core.exceptions.FieldError), since the problem is the primary key is not set.

Another possibility is that `assert self.pk is not None` would be sufficient - is it reasonable to expect this will only occur if the programmer using conman has made a mistake?

Otherwise, perhaps one of `TypeError`, `ValueError` or `RuntimeError`, but I'm not thrilled by these.

I'm currently leaning towards the `assert`, but I can be convinced otherwise.